### PR TITLE
[DD-502] Remove Datadog Synthetics from our CI Workflow

### PR DESCRIPTION
**Pull request by tshen815**
[Ticket](https://circleci.atlassian.net/browse/DD-502)
Original PR: #5881 
Pipeline for preview jobs: https://app.circleci.com/pipelines/github/circleci/circleci-docs?branch=DD-502%2Fremove-DDRum-from-CI-preview&filter=all

# Changes

- remove datadog synthetics CI script
- remove `datadog-synthetics-ci` job from our CI workflow
- remove dependencies for datadog-synthetics-ci

# Description
We want to remove Datadog synthetics from our CI as it has been causing more problems than it is solving and we have other systems in place to make sure we are not breaking anything. 

# Considerations
For more context please refer to this thread: https://circleci.slack.com/archives/C021NCSEM44/p1647987134403809

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -985,21 +980,21 @@
     tslib "^1.10.0"
 
 "@datadog/datadog-ci@^0.17.8":
-  version "0.17.8"
-  resolved "https://registry.yarnpkg.com/@datadog/datadog-ci/-/datadog-ci-0.17.8.tgz#285b749c9afe23e63bab54e8d3edd4d87dda8add"
-  integrity sha512-9Y44r+4fdGIhxs8sUtSkKs2cv5JQh8OcTJL8pxZe9vslhhQKMr9YZ7UF+ms0+G3ZfmV5lf6N+zoFeglnMd/UwA==
+  version "0.17.13"
+  resolved "https://registry.yarnpkg.com/@datadog/datadog-ci/-/datadog-ci-0.17.13.tgz#cc7f43cb8fb165983bddce485758dd5476bbd566"
+  integrity sha512-2wQtmAR+Nk43HsJJX0QVwuJ67dRuExA84gdHwsWSiFJF+0CDUcRx8Tl6dCy7sP5RAUYF/3rDW5DVTt7FcXUDEw==
```
Comment (rsylvian) [inline yarn.lock:985]: how come this changed if nothing changed in the package.json?

**Comment by rsylvian** [comment]
Can we try that `-preview` is also not broken?

**Review by chantelle11 — APPROVED** [review]
LGTM!!!

File: yarn.lock (hunk 1 above)
Comment (tshen815) [reply yarn.lock]: reverted `yarn.lock` back to master branch 👍

**Review by rsylvian — APPROVED** [review]

**Pull request merged**